beep/structure.py

```diff
@@ -1773,6 +1800,23 @@ def from_dict(cls, d):
         return cls(data, metadata)
 
 
+def determine_whether_step_is_waveform(step_dataframe):
```

Comment:
Just to be clear, this will only work on maccor? Checking for monotonicity in voltage or capacity does not work? I guess I worry about creating methods this far into pipeline that only work with specific cycler makes since it means we have more to keep track of if/when we have waveforms for Biologic/Arbin

Reply:
For now, it only works for maccor, but can easily add switches for other cyclers if we have sample data.

Reply:
We would need a threshold or some heuristics to determine when a cycle with some non-monotonicities is determined to be waveform vs not.